miasm2/analysis/depgraph.py

```diff
@@ -374,6 +374,37 @@ def emul(self, step=False):
                 for depnode in self.input}
 
 
+def follow_simp_expr(exprs):
+    """Avoid tracking useless elements, as XOR EAX, EAX"""
+
+    follow = set()
+    for expr in exprs:
+        follow.add(expr_simp(expr))
+    return follow, set()
+
+def follow_mem_cst(exprs):
+    follow = set()
+    for expr in exprs:
+        follow.update(expr.get_r(mem_read=True, cst_read=True))
+    return follow, set()
+
+def follow_nomem_cst(exprs):
```

Comment:
`follow_nomem_cst` implementation is very close to `follow_mem_cst`. You could define a function which just take in argument `mem_read` and `cst_read` values and merge them.